lib/taiko.js

```diff
@@ -2220,20 +2221,41 @@ module.exports.dismiss = async () => {
  * loadPlugin(IDm pluginHandler);
  *
  * @param {string} ID - unique id or name of the plugin
- * @param {Function} clientHandler - callback method to set taiko instance for plugin
+ * @param {Function} clientHandler - DEPRICATED use init method instead
+ * @param {Function} init - callback method to set taiko instance for plugin
  */
 
-module.exports.loadPlugin = (id, clientHandler) => {
+
+const loadPlugin = (id, init) => {
     try {
         if (!plugins.has(id)) {
-            if (client !== undefined) clientHandler(module.exports);
-            plugins.set(id, clientHandler);
+            init(module.exports, xhrEvent);
+            plugins.set(id, init);
         }
     } catch (error) {
         console.trace(error);
     }
 };
 
+module.exports.loadPlugin = loadPlugin;
+
+
+const overriddenAPIs = {};
+getPlugins().forEach( (pluginName) => {
+    let plugin = require(path.resolve(`node_modules/${pluginName}`));
+    loadPlugin(plugin.ID, plugin.init);
+    module.exports[plugin.ID] = plugin;
```

Comment:
plugins need not have IDs as taiko plugins will follow the convention of having name as `taiko-<pluginName>` pluginName has to be unique thus the same can be used to avoid conflicts between pluginIds and the exported object should have key as `pluginName` if different Id is set that convention will break